### PR TITLE
Cargo.toml: specify revisions for all git dependencies

### PR DESCRIPTION
**Pull request by andylokandy**
## What have you changed? (mandatory)

- Add `rev=$sha` for each git dependencies. Currently all git revisions are held by `Cargo.lock`, which might be changed unexpectedly by `cargo update`.

## What are the type of the changes? (mandatory)

- Misc (other changes)

## How has this PR been tested? (mandatory)

Run passed `cargo test`

## Does this PR affect documentation (docs) or release note? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Refer to a related PR or issue link (optional)

https://github.com/tikv/tikv/issues/4283

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -117,14 +117,13 @@ vlog = "0.1.4"
 twoway = "0.2.0"
 cop_datatype = { path = "components/cop_datatype" }
 panic_hook = { path = "components/panic_hook" }
-tipb = { git = "https://github.com/pingcap/tipb.git" }
-kvproto = { git = "https://github.com/pingcap/kvproto.git" }
+tipb = { git = "https://github.com/pingcap/tipb.git", rev = "833c2ffd" }
```
Comment (breeswish) [inline Cargo.toml]: You can use the `[patch]` section for `tipb`, `kvproto` and `rocksdb`. Otherwise many files need to be updated to update a single dep.

Comment (andylokandy) [reply Cargo.toml]: Because of https://github.com/rust-lang/cargo/issues/5478 it's currently not possible.

Comment (breeswish) [reply Cargo.toml]: Then I think we should wait it to be resolved, since this PR requires us to update so many fragmentary `Cargo.toml` frequently.

Comment (andylokandy) [reply Cargo.toml]: Since it seems to be long term waiting, I'm going to close this PR.

**Pull request closed without merging**